core-c-bridge/include/temporal-sdk-core-c-bridge.h

```diff
@@ -812,6 +959,32 @@ void temporal_core_complete_async_reserve(void *sender, uintptr_t permit_id);
 void temporal_core_set_reserve_cancel_target(struct TemporalCoreSlotReserveCtx *ctx,
                                              void *token_ptr);
 
+/**
+ * Create a new [Worker] with a custom callback-backed [temporal_sdk_core::WorkerClient] implementation.
+ *
+ * # Safety
+ * - All pointer args (`client_rpcs`, `options`, `client_config`, `runtime`) must be non-null and valid.
+ * - `system_info_ref.data` must point to at least `system_info_ref.size` bytes if non-zero.
+ * - Caller retains ownership; these must outlive the call.
+ *
+ * # Params
+ * - `client_rpcs`: pointer to [CallbackBasedWorkerClientRPCs] implementing Core SDK [temporal_sdk_core::WorkerClient] RPCs.
+ * - `options`:     pointer to [WorkerOptions].
+ * - `client_config`: pointer to [CallbackBasedWorkerClientConfig] (client name, version, identity).
+ * - `system_info_ref`: zero-copy [GetSystemInfoResponse] bytes or empty.
+ * - `runtime`:        pointer to [Runtime].
+ *
+ * # Returns
+ * A [WorkerOrFail] struct containing:
+ * - `worker`: on success, a heap-allocated [Worker] pointer; null on error.
+ * - `fail`: on error, a C string (UTF-8) describing the failure; null on success.
+ */
+struct TemporalCoreWorkerOrFail temporal_core_new_callback_based_worker_client(const struct TemporalCoreCallbackBasedWorkerClientRPCs *client_rpcs,
```

Comment:
```suggestion
struct TemporalCoreWorkerOrFail temporal_core_callback_based_worker_new(const struct TemporalCoreCallbackBasedWorkerClientRPCs *client_rpcs,
```

or

```suggestion
struct TemporalCoreWorkerOrFail temporal_core_worker_callback_based_new(const struct TemporalCoreCallbackBasedWorkerClientRPCs *client_rpcs,
```

Pedantic, but worth being clear this is creating a _worker_ not a _client_